### ISSUE TITLE
[Accessibility Bug] CardElement aria-hidden component contains focusable elements

### ISSUE THREAD

**Issue by lmcphetridge** [stripe/react-stripe-js#236]
Feature request or idea? Consider opening an
[API review](https://github.com/stripe/react-stripe-js/tree/master/.github/API_REVIEW.md)!

### Summary

Hi! We're working on the accessibility of our site and noticed a violation that originates in the `CardElement` component. Using this component results in an input field being added to the DOM with an `aria-hidden` attribute set to true but is still focusable. I think the fix in this case is to also add `tabIndex = -1` when `aria-hidden` is true.

### Other information

For added context below is a snippet of the error messaging:

**Issue Description**
Ensures aria-hidden elements do not contain focusable elements

**Element location**
`.__PrivateStripeElement-input`

**Element source**
`<input class="__PrivateStripeElement-input" aria-hidden="true" aria-label=" " autocomplete="false" maxlength="1" style="border: none !important; display: block !important; position: absolute !important; height: 1px !important; top: -1px !important; left: 0px !important; padding: 0px !important; margin: 0px !important; width: 100% !important; opacity: 0 !important; background: transparent !important; pointer-events: none !important; font-size: 16px !important;">`

**To solve this issue, you need to...**
Fix the following:
Focusable content should be disabled or be removed from the DOM

**Comment by christopher-stripe** [comment]
Hi @lmcphetridge, I believe this warning is a false positive relating to how we manage focus between frames in Stripe.js.

When you render a `<CardElement />`, the resulting DOM structure looks like this (simplified):

```html
<div class="StripeElement">
  <div class="__PrivateStripeElement>
    <iframe src="https://js.stripe.com/..."></iframe>
    <input class="__PrivateStripeElement-input" aria-hidden="true">
  </div>
</div>
```

The actual inputs for entering card information are rendered inside the `<iframe>`, but are intended to be embedded alongside the more normal inputs in your payment form that you manage. In some browsers, tabbing from an input in your frame would not automatically tab into the inputs inside the `<iframe>`—the iframed inputs would be entirely skipped over. This prevents users who navigate with the keyboard (either by preference or due to motor disability) from being able to enter their details.

So we add the visually hidden `.__PrivateStripeElement-input` element which, when it receives focus, will forward focus into the inputs inside the `<iframe>`. This makes the iframed-card inputs reachable by keyboard alone across browsers. Setting `tabIndex="-1"` would prevent this solution, because the hidden `.__PrivateStripeElement-input` input would never be focused. We _do_ however want the `.__PrivateStripeElement-input` input to be hidden from screen readers, which can access the iframed-inputs directly.

**Comment by lmcphetridge** [comment]
Thanks for the response @christopher-stripe. I think I understand what's happening and see why simply altering the tabIndex could be a problem. 

Would it be possible to address this by adding a `hidden` attribute in combination to `aria-hidden`  or would that still be an issue for browser compatibility? I see it listed as a best practices section here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute . This solution also passes the check on my side.

**Comment by TechThomas** [comment]
Can this come back?

**Comment by dylanb** [comment]
There is no reason to hide the input from the screen reader. if your focus is being managed correctly, when the focus shifts from the hidden input to the iframe, the screen reader focus should follow it. I don't understand the need for the aria-hidden attribute.

**Comment by maggievcody** [comment]
Can we bring this back?

**Comment by renato-bohler** [comment]
+1 to bring this back.

My team might be blocked from turning on automated accessibility checks on a project pipeline because of this single third-party violation.

**Comment by saritaSchen** [comment]
I also agree that this needs to be tackled!
According to the European Accessibility Act, at the latest before June 2025! 🚨
